Corona-Warn-App/src/main/java/de/rki/coronawarnapp/datadonation/safetynet/SafetyNetClientWrapper.kt

```diff
@@ -0,0 +1,113 @@
+    private val safetyNetClient: SafetyNetClient,
+    private val environmentSetup: EnvironmentSetup
+) {
+
+    suspend fun attest(nonce: ByteArray): Report {
+        val jwsResult = callClient(nonce).jwsResult ?: throw SafetyNetException(
+            Type.ATTESTATION_FAILED,
+            "JWS was null"
+        )
+
+        val components = jwsResult.split(".")
+        if (components.size != 3) throw SafetyNetException(
+            Type.ATTESTATION_FAILED,
+            "Invalid JWS: Components are missing."
+        )
+
+        val header = try {
+            components[0].decodeBase64Json()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS header.", e)
+        }
+        val body = try {
+            components[1].decodeBase64Json()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS body.", e)
+        }
+        val signature = try {
+            components[2].decodeBase64()!!.toByteArray()
+        } catch (e: Exception) {
+            throw SafetyNetException(Type.ATTESTATION_FAILED, "Failed to decode JWS signature.", e)
+        }
+
+        return Report(
+            jwsResult = jwsResult,
+            header = header,
+            body = body,
+            signature = signature
+        )
+    }
+
+    private fun String.decodeBase64Json(): JsonObject {
+        val rawJson = decodeBase64()!!.string(Charsets.UTF_8)
+        return JsonParser.parseString(rawJson).asJsonObject
+    }
+
+    private suspend fun callClient(nonce: ByteArray): SafetyNetApi.AttestationResponse = suspendCoroutine { cont ->
+        safetyNetClient.attest(nonce, environmentSetup.safetyNetApiKey)
+            .addOnSuccessListener {
+                Timber.tag(TAG).v("Attestation finished with %s", it)
+                cont.resume(it)
+            }
+            .addOnFailureListener {
+                Timber.tag(TAG).w(it, "Attestation failed.")
+                val wrappedError = if (it is ApiException && it.statusCode == CommonStatusCodes.NETWORK_ERROR) {
+                    SafetyNetException(Type.ATTESTATION_REQUEST_FAILED, "Network error", it)
+                } else {
+                    SafetyNetException(Type.ATTESTATION_FAILED, "SafetyNet client returned an error.", it)
+                }
+                cont.resumeWithException(wrappedError)
+            }
+    }
+
+    data class Report(
+        val jwsResult: String,
+        val header: JsonObject,
+        val body: JsonObject,
+        val signature: ByteArray
+    ) {
+        val nonce: String? = body.get("nonce")?.asString
+
+        val apkPackageName: String? = body.get("apkPackageName")?.asString
+
+        val basicIntegrity: Boolean = body.get("basicIntegrity")?.asBoolean == true
+        val ctsProfileMatch = body.get("ctsProfileMatch")?.asBoolean == true
+
+        val evaluationTypes = body.get("evaluationType")?.asString
```

Comment:
This also handles the scenario where `evaluationType` is not set at all, right?

Reply:
Correct, it would fallback to `emptyList()`.
I have added additional test cases to check that.